transport/TcpServerEndpoint.h

```diff
@@ -24,18 +25,19 @@ class TcpServerEndpoint : public ServerEndpoint, public RtcePoll::IEventListener
         RtcePoll& rtcePoll,
         size_t maxPengingSessions,
         TcpServerEndpoint::IEvents* defaultListener,
+        TcpEndpointFactory& tcpEndpointFactory,
```

Comment:
See similar comment: Can it be passed/owned as shared_ptr instead of raw ptr? Why passing differently?